extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/runtime/AbstractJpaOperations.java

```diff
@@ -0,0 +1,506 @@
+package io.quarkus.hibernate.reactive.panache.common.runtime;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import javax.enterprise.util.TypeLiteral;
+import javax.persistence.LockModeType;
+import javax.persistence.PersistenceException;
+import javax.transaction.SystemException;
+import javax.transaction.TransactionManager;
+
+import org.hibernate.internal.util.LockModeConverter;
+import org.hibernate.reactive.mutiny.Mutiny;
+import org.hibernate.reactive.mutiny.Mutiny.Session;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.panache.common.Parameters;
+import io.quarkus.panache.common.Sort;
+import io.quarkus.panache.common.exception.PanacheQueryException;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+public abstract class AbstractJpaOperations<PanacheQueryType> {
+
+    protected abstract PanacheQueryType createPanacheQuery(Uni<Mutiny.Session> em, String query, String orderBy,
+            Object paramsArrayOrMap);
+
+    protected abstract Uni<List<?>> list(PanacheQueryType query);
+
+    protected abstract Multi<?> stream(PanacheQueryType query);
+
+    //
+    // Instance methods
+
+    public Uni<Void> persist(Object entity) {
+        return getEntityManager().flatMap(em -> persist(em, entity));
+    }
+
+    public Uni<Void> persist(Mutiny.Session em, Object entity) {
+        if (!em.contains(entity)) {
+            return em.persist(entity).map(v -> null);
```

Comment:
What about this:
```
		return em.persist( entity ).onItem().ignore().andContinueWithNull();
```

Reply:
Yeah, I've found the `map(v -> null)` to be a lot shorter and to the point. I'd rather keep this :)

Reply:
I first use `map(v -> null)` than Clement said that `.onItem().ignore().andContinueWithNull()` is better, I don't remember why (maybe it provides the item to be emitted ...)